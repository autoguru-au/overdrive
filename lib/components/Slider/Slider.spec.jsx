import React from 'react';
import { render } from '@testing-library/react';
import { Slider } from '.';

describe('<Slider />', () => {
	it('should not throw', () => {
		expect(() => render(<Slider />)).not.toThrow();
	});

	it('should match the snapshot', () => {
		expect(render(<Slider />).container.firstChild).toMatchSnapshot();
	});
});
