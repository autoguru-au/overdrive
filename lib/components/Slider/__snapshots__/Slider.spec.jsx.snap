// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Slider /> > should match the snapshot 1`] = `
<div
  class="theme_OD_Base__j49br80"
  data-od-component="provider"
>
  <div
    class="elementReset_resetVariants__18vhvlid elementReset_resetVariants_as_div__18vhvlie elementReset_block__18vhvli4 Slider__zjmk6a0 sprinkles_display_flex_mobile__onxwju6l sprinkles_flexDirection_column_mobile__onxwjupx sprinkles_gap_3_mobile__onxwjuid sprinkles_width_full_mobile__onxwjuh5"
    data-od-component="slider"
    id="react-aria-«r6»"
    role="group"
  >
    <div
      class="Slider__zjmk6a2 sprinkles_display_flex_mobile__onxwju6l sprinkles_justifyContent_center_mobile__onxwjuoh sprinkles_marginBottom_3_mobile__onxwjuxx"
    >
      <output
        aria-live="off"
        class="Slider__zjmk6a3 sprinkles_fontSize_7_mobile__onxwjual sprinkles_fontWeight_semiBold__onxwju5l sprinkles_color_normal__onxwju2s"
        for="react-aria-«r6»-0"
      >
        0
      </output>
    </div>
    <div
      class="Slider__zjmk6a4 sprinkles_display_flex_mobile__onxwju6l sprinkles_alignItems_center_mobile__onxwjumt sprinkles_gap_4_mobile__onxwjuih sprinkles_width_full_mobile__onxwjuh5"
    >
      <span
        class="elementReset_resetVariants__18vhvlid elementReset_resetVariants_as_span__18vhvlim  sprinkles_display_block_mobile__onxwju6d Icon_makeResponsiveStyle_key_bp__1gy3ire4"
        data-od-component="icon"
      >
        <svg
          aria-hidden="true"
          class="elementReset_resetVariants__18vhvlid elementReset_resetVariants_as_div__18vhvlie elementReset_block__18vhvli4 sprinkles_display_block_mobile__onxwju6d sprinkles_width_full_mobile__onxwjuh5 sprinkles_height_full_mobile__onxwjufl"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 12.998H5V11l14-.002v2z"
          />
        </svg>
      </span>
      <div
        class="Slider__zjmk6a5 sprinkles_position_relative_mobile__onxwju8h sprinkles_width_full_mobile__onxwjuh5 sprinkles_display_flex_mobile__onxwju6l"
      >
        <div
          class="Slider_track__zjmk6a7 Slider__zjmk6a6 sprinkles_borderRadius_pill_mobile__onxwjuc9 sprinkles_height_1_mobile__onxwjue9 sprinkles_position_relative_mobile__onxwju8h sprinkles_width_full_mobile__onxwjuh5"
          style="position: relative;"
        >
          <div
            class="Slider_thumb__zjmk6a9 Slider__zjmk6a8 sprinkles_borderBottomWidth_none_mobile__onxwjud9 sprinkles_borderLeftWidth_none_mobile__onxwjuct sprinkles_borderRightWidth_none_mobile__onxwjudp sprinkles_borderTopWidth_none_mobile__onxwjue5 sprinkles_width_5_mobile__onxwjug9 sprinkles_height_5_mobile__onxwjuep sprinkles_borderRadius_full_mobile__onxwjucd"
            style="position: absolute; left: 0%; transform: translate(-50%, -50%);"
          >
            <div
              class="elementReset_resetVariants__18vhvlid elementReset_resetVariants_as_div__18vhvlie elementReset_block__18vhvli4 VisuallyHidden_visuallyHidden__12d73qi0"
              data-od-component="visually-hidden"
            >
              <input
                aria-describedby=""
                aria-details=""
                aria-labelledby="react-aria-«r6»"
                aria-orientation="horizontal"
                aria-valuetext="0"
                id="react-aria-«r6»-0"
                max="100"
                min="0"
                step="5"
                tabindex="0"
                type="range"
                value="0"
              />
            </div>
          </div>
        </div>
      </div>
      <span
        class="elementReset_resetVariants__18vhvlid elementReset_resetVariants_as_span__18vhvlim  sprinkles_display_block_mobile__onxwju6d Icon_makeResponsiveStyle_key_bp__1gy3ire4"
        data-od-component="icon"
      >
        <svg
          aria-hidden="true"
          class="elementReset_resetVariants__18vhvlid elementReset_resetVariants_as_div__18vhvlie elementReset_block__18vhvli4 sprinkles_display_block_mobile__onxwju6d sprinkles_width_full_mobile__onxwjuh5 sprinkles_height_full_mobile__onxwjufl"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 12.998h-6v6h-2v-6H5v-2h6v-6h2v6h6v2z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
